### PR TITLE
MorseCode Decoder

### PR DESCRIPTION
**Pull request by inesmaio**
New feature.
Decoder from morse code into words and the other way around.
ESLint checked
Jest tests

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -17,5 +17,6 @@
     },
     "rules": {
         "no-plusplus": "off"
+
```
Comment (bcotrim) [inline .eslintrc.json]: Extra white line for no reason.
Seems like leftover form a change, I think you can delete this.

File: morseCode/morseCodeToWords.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  const inputArr = input.split(SEPARATOR);
+  const coderMorseArr = inputArr.map((element) => (element.split(' ')));
+  return coderMorseArr;
+};
+
+// Transform Morse code into words
+const morseCodeToWords = (arr) => {
+  let result = [];
+  const morseKeys = Object.keys(morseCode);
+  arr.map((elem) => {
```
Comment (bcotrim) [inline morseCode/morseCodeToWords.js]: I think `reduce` would be a better option for this.

File: morseCode/morseCodeToWords.test.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const { mCTWordsResult } = require('./morseCodeToWords');
```
Comment (bcotrim) [inline morseCode/morseCodeToWords.test.js]: We are missing some test scenarios (null input for example)

Also it would be better to throw an error when getting an `undefined` translation instead of returning that result.

File: morseCode/wordsToCodeMorse.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  const SEPARATOR = ' ';
+  const inputArr = input.split(SEPARATOR);
+  const wordsArr = inputArr.map((element) => (element.split('')));
+  return wordsArr;
+};
+
+// Decode input into morseCode
+const wordsToMorseCode = (arr) => {
+  let result = [];
+  arr.map((elem) => {
```
Comment (bcotrim) [inline morseCode/wordsToCodeMorse.js]: Same as before I think it would be better to use `reduce` for this case.

File: morseCode/validateInput.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+const { wordsToMCResult } = require('./wordsToCodeMorse');
+const { mCTWordsResult } = require('./morseCodeToWords');
+const { InputError } = require('../utils/errors');
+const { letters } = require('../utils/variables');
+
+
+const inputValidation = (userInput) => {
```
Comment (bcotrim) [inline morseCode/validateInput.js]: This is not only doing input validation.
It's also calling the appropriate function to solve the problem.

I think the function is good, just the name shouldn't be `inputValidation` in my opinion.

**Review by bcotrim — CHANGES_REQUESTED** [review]

File: morseCode/morseCodeToWords.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
+const { morseCode } = require('../utils/variables');
+
+// Transform the input into an Array
+const arrMorseCodeToWords = (input) => {
+  const SEPARATOR = '   ';
+  const inputArr = input.split(SEPARATOR);
+  const coderMorseArr = inputArr.map((element) => (element.split(' ')));
```
[the remark this answers is not in the document]
Comment (inesmaio) [reply morseCode/morseCodeToWords.js]: makes sence, thanks for the feedback

File: morseCode/validateInput.js (hunk 5 above)
Comment (inesmaio) [reply morseCode/validateInput.js]: You're right so, I changed the output of `inputValidation()`  to true or false.
This way I can only test the input.

Comment (inesmaio) [reply morseCode/validateInput.js]:
```suggestion
const inputValidation = (input) => {
  let validation;
  const REGEX = /[.-]/gm;
  if (input.search(REGEX) >= 0) {
    if (input.search(LETTERS) >= 0) {
      throw new InputError('Insert a valid Morse Code. No letter or numbers allowed.');
    } else {
      validation = true;
      return validation;
    }
  } else {
    validation = false
    return validation;
  }
};
```

File: morseCode/morseCodeToWords.test.js (hunk 3 above)
Comment (inesmaio) [reply morseCode/morseCodeToWords.test.js]: The test case that returned undefined was removed, that it's not a real scenario, since the only way to get errors is if the input is incorrect, and I build a function to check the input before go through the decode execution.
